routes/email-verification.js

```diff
@@ -0,0 +1,54 @@
+let express = require('express');
+let router =  express.Router();
+let db = require('../db.js');
+let auth = require('../auth.js');
+let url = require('url');
+
+router.get('/', function(req, res, next) {
+    res.render('email-verification');
+});
+
+router.post('/resend', function(req, res, next) {
+    // Resend verification email.
+    let email = req.body.email;
+    db.EmailVerification.resendVerificationEmail(email, function(err, userFound) {
+        if (err) {
+            res.render('error', {message: 'Resending verification email FAILED! :C'});            
+        }
+        if (userFound) {
+            res.render('email-verification', {statusmessage: 'An email has been sent to you again at ' + email + '. Please check it to verify your account.'});            
+        } else {
+            res.render('email-verification', {errormessage: 'Your verification code has expired. Please sign up again.'});                        
+        }
+    });
+});
+
+// user accesses the link that is sent
+router.get('/:URL', function(req, res) {
+    let url = req.params.URL;
+
+    db.EmailVerification.confirmTempUser(url, async function(err, user, codes) {
+        if (user) {
+            // Authenticate and enroll in courses.
+            auth.sendCookie(res, user._id);
+            try {
+                if (codes) {
+                    // Add user to each course they provided.
```

Comment:
This would be unnecessary if you just use isVerified flag, right? Basically all this entire endpoint would do is toggle that flag to true. I'm concerned about the indepotency of this function - what happens if the user accesses it many times? With a simple flag toggle to "true", it won't matter and it is also easier to reason about.